### DISCUSSION TITLE
Minify Issue when installing on Termux (Android)

### DISCUSSION THREAD

**Discussion by PolarisFx** [lncrawl/lightnovel-crawler#1744] [Q&A] [answered]
I can't get the new version to install in Termux. I've been trying for awhile but no matter what I do, it always installs 2.34.0. I've started from scratch a dozen or so times, tried forcing different versions but 2.34 is the only version that will install without errors. Forcing 3.0.1 via pip gives this error:

![Termux-Capture](https://user-images.githubusercontent.com/12600197/198332937-8517c1ca-0d9d-426d-82ab-e1b46fa57795.PNG)

**Comment by PolarisFx** [comment, marked as answer]
Needed to install Rust, binutils and build minify-html from source. Got 3.0.1 installed.

**Reply by budikesuma** [reply]
> Needed to install Rust, binutils and build minify-html from source. Got 3.0.1 installed.

@PolarisFx 
How?
I still got errors.

**Reply by PolarisFx** [reply]
pip will try to use the wheel version if available, if not you can build from source. There doesn't appear to be a wheel version for aarm64. So you need to build it from source using:

pip install minify-html --no-binary :all:

You'll need rust, and the android libraries from binutils to compile, and it'll take a bit of time, but after you get it installed lncrawl will install with no issues. 

If it's not installing after that make sure you have everything else you need:

libxml2 libxslt libjpeg-turbo wheel setuptools

I followed the tutorial on the main lncrawl page, did all the initial setup. Then when it showed it was missing minify-html I eventually learned how to build from source and it worked from there.

**Comment by budikesuma** [comment]
@PolarisFx 

Too confused, still not much help.
I only managed to install `rust` (not `Rust`) and `binutils` after a git clone, with no luck with minify-html, by the way `--no-binary` is not recognized by termux.

**Reply by PolarisFx** [reply]
Well... then I have no idea. I would love to say I know what I'm doing, but I'm not a Python guy. I did get it working:

![Termux-Capture3](https://user-images.githubusercontent.com/12600197/198905479-6f759850-3ba8-4e2c-bc0b-5fbb7764d3de.PNG)

And I did get it working by compiling it from source, using pip after installing rust & binutils. But to be honest, I have no clue how I forced pip to use the source instead of the wheel version. When you asked me, I googled the problem and found the --no-binary thing so I thought thats what I did, but I just found out by trying to run it that it's been deprecated.

**Comment by budikesuma** [comment]
Finally was able to install it into termux, actually no need to install from source.

There is only a problem downloading the WEBP type image. (Not supported by termux).
.
.
![Screenshot_20221101-074136_Termux](https://user-images.githubusercontent.com/50254552/199135156-4892f5bb-79ef-4eb6-98f5-a117dd74b158.jpg)

![Screenshot_20221101-074202_Termux](https://user-images.githubusercontent.com/50254552/199135207-e0f90526-2f79-464f-af79-8fc985eef1ea.jpg)

![Screenshot_20221101-074218_Termux](https://user-images.githubusercontent.com/50254552/199135234-91b57451-0700-4009-96c1-7190b0f6b6c5.jpg)

![Screenshot_20221101-074006_Termux](https://user-images.githubusercontent.com/50254552/199135271-682cec79-54bc-4875-b802-c9656e81ff5f.jpg)

**Comment by budikesuma** [comment]
WEBP issue solved by these steps:


1. Uninstall Pillow module (`pip uninstall Pillow`)
2. Clear pip cache (`pip cache purge`).
3. `pkg install libwebp` (omit this step if already done).
4. Install Pillow again (`pip install Pillow==9.2.0`). Done!

But then a new problem arises when there are `.jpg` images from mirror sites such as `wordpress.com`

Try downloading `chapter 389` of the novel `Emperor's Domination`, after you have solved the WEBP issue above.

Surprisingly the same `.jpg` image from another mirror site can be downloaded without any problem as from the mirror site `nexusmods.com` in `chapter 390`.
.

### Test link: https://www.wuxiaworld.com/novel/emperors-domination 

.
.
![Screenshot_20221101-090600_Termux](https://user-images.githubusercontent.com/50254552/199143781-09e76eaf-e290-475d-b688-5799c4edde5f.jpg)

![Screenshot_20221101-090618_Termux](https://user-images.githubusercontent.com/50254552/199143825-7e692f38-ad66-4512-86f1-1e21484ff248.jpg)

![Screenshot_20221101-090629_Termux](https://user-images.githubusercontent.com/50254552/199143845-ec35cc61-5f8f-403e-b62a-9e5e22b3391d.jpg)
